source/develop/release-management/features/external-network-provider.html.md

```diff
@@ -0,0 +1,376 @@
+
+* the engine will try to locate the port on the external provider (GET ports)
+* the engine will delete the port from the external provider (DELTE ports)
+
+#### Migrate a VM to another host
+
+When a VM is migrated to another host the following actions are executed:
+
+* the VIF driver, invoked by the VDSM before_migration_source disconnects the NIC from the external network on the source VDSM
+* the VIF driver, invoked by the VDSM before_migration_destination connects the NIC ti\o the external network on the destination VDSM
+
+## External networks REST API
+
+The communication between the engine and the external network provider is done using a REST API based on this of OpenStack neutron. Only a subset of the Neutron API is used.
+
+The REST API uses the following operations:
+
+Required tokens:
+headers={Content-Type=[application/json], x-openstack-request-id=[<authentication token>]}
+
+Authentication (optional):
+
+* authenticate: POST: http://host:35357/v2.0/tokens
+
+Request:
+{"auth": {"passwordCredentials": {"username": "<user>", "password": "<password>"}}}
+Response: {
+"access":{
+    "token":{
+        "id": "<token>"
+    }
+}}
+
+
+* tets connection: GET: http://host:9696/v2.0/
+
+Networks:
+
+* network details: GET: http://host:9696/v2.0/network/<network id>
+
+* list networks: GET: http://host:9696/v2.0/networks
+
+* delete a network: DELETE: http://host:9696/v2.0/networks
+
+Ports:
+
+* list ports: GET: http://host::9696/v2.0/ports   (Check if port exists)
+
+* port details:  GET: http://host::9696/v2.0/ports/<port id>
+
+* add a port: POST:  http://host::9696/v2.0/ports/<port id>
+
+* delete port: DELETE:  http://host::9696/v2.0/ports/<port id>
+
+Subnets:
+
+* list subnets: GET: http://host:9696/v2.0/subnets
+
+* adding a subnet: POST: http://host:9696/v2.0/subnets
+
+* deleting a subnet: DELETE: http://host:9696/v2.0/network/<network id>
+
+## VIF driver (prepared by the external network provider)
+
+The "virtual interface driver" will be a set of scripts provided by the external network provider which will facilitate the network operations related to an external network on a host (VDSM).
+The operations include:
+
+* setting up the external network on the host
+
+* creating and attaching a VM NIC to an external network
+
+* deleteing and detaching a VM NIC from an external network
+
+* hotplugging a NIC
+
+* unplugging a NIC
+
+* VM migration
+
+* reporting the external network and it's relations to any host NIC to the engine
+
+* reporting VM NIC statistics to the engine
+
+The VIF driver will be notified about any life-cycle events relevant to external networks by means of the VDSM "hooks".
+VDSM Hooks are a means to insert arbitrary commands and scripts at certain point in a VM's lifecycle as well as in VDSM daemon's lifecycle.
+
+The following hooks are relevant to the external networks:
+
+#### before_device_create/after_device_create
```

Comment:
Somewhere here we need to discuss waiting for provider to give us the ack to run the VM once the network config is generated on the host and how this is done. This is to prevent race between VM boot and network setup.